Sources/SwiftProtobuf/Message+JSONAdditions.swift

```diff
@@ -93,3 +93,72 @@ public extension Message {
   }
 }
 
+
+public extension Array where Iterator.Element: Message {
```

Comment:
We should not provide interoperability with `JSONSerialization`. Our library is an *implementation of protocol buffers*, which has a very specific JSON format. Supporting anything outside that is out of scope.

However, the case in this PR seems allowable because we're talking about an array of objects that have themselves been encoded to protobuf JSON spec. The spec doesn't say anything about that, so we have more flexibility here. If it's useful to people writing APIs that work with arrays of objects, that arguably has value.

On this specific issue (extensions), I'm leaning toward the static methods. I'm not *crazy* about it, but it sounds like a better option.

Rationale: If I were designing a library whose *entire purpose* was to add extensions to Swift's common types, I'd feel differently—people who didn't want those extensions would just not use the library. But since we're a project that aims to provide specific functionality that doesn't currently do that, it seems like the right choice here is to not force extensions on users that they can't separate from the rest of the library. (If we had submodules that weren't implicitly exported to their parent, we could do something nice here, but we don't have submodules at all yet.)